test/unit/logic/delegate.js

```diff
@@ -0,0 +1,534 @@
+	nameexist: 1
+};
+
+var senderHash = crypto.createHash('sha256').update(validSender.secret, 'utf8').digest();
+var senderKeypair = ed.makeKeypair(senderHash);
+
+var validTransaction = {
+	type: 2,
+	amount: 0,
+	fee: 0,
+	timestamp: 0,
+	recipientId: null,
+	senderId: '10881167371402274308L',
+	senderPublicKey: 'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+	asset: {
+		delegate: {
+			username: 'genesis_100',
+			publicKey: 'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9'
+		}
+	},
+	signature: '5495bea66b026b0d6b72bab8611fca9c655c1f023267f3c51453c950aa3d0e0eb08b0bc04e6355909abd75cd1d4df8c3048a55c3a98d0719b4b71e5d527e580a',
+	id: '8500285156990763245'
+};
+
+var rawValidTransaction = {
+	t_id: '8500285156990763245',
+	b_height: 1,
+	t_blockId: '6524861224470851795',
+	t_type: 2,
+	t_timestamp: 0,
+	t_senderPublicKey: 'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+	m_recipientPublicKey: null,
+	t_senderId: '10881167371402274308L',
+	t_recipientId: null,
+	t_amount: '0',
+	t_fee: '0',
+	t_signature: '5495bea66b026b0d6b72bab8611fca9c655c1f023267f3c51453c950aa3d0e0eb08b0bc04e6355909abd75cd1d4df8c3048a55c3a98d0719b4b71e5d527e580a',
+	t_SignSignature: null,
+	t_signatures: null,
+	confirmations: 284,
+	d_username: 'genesis_100'
+};
+
+describe('delegate', function () {
+
+	var transactionMock;
+	var accountsMock;
+	var delegate;
+
+	var trs;
+	var rawTrs;
+	var sender;
+
+	function dummyCb () {};
+
+	before(function () {
+		transactionMock = sinon.mock({});
+	});
+
+	beforeEach(function () {
+		trs = _.cloneDeep(validTransaction);
+		rawTrs = _.cloneDeep(rawValidTransaction);
+		sender = _.cloneDeep(validSender);
+
+		accountsMock = {
+			setAccountAndGet: sinon.mock(),
+			getAccount: sinon.mock()
+		};
+
+		delegate = new Delegate(modulesLoader.scope.schema);
+		delegate.bind(accountsMock);
+	});
+
+	describe('constructor', function () {
+
+		it('should be attach schema to library variable', function () {
+			new Delegate(modulesLoader.scope.schema);
+			var library = Delegate.__get__('library');
+
+			expect(library).to.eql({
+				schema: modulesLoader.scope.schema
+			});
+		});
+	});
+
+	describe('bind', function () {
+
+		it('should attach empty object to private modules.accounts variable', function () {
+			delegate.bind({});
+			var modules = Delegate.__get__('modules');
+
+			expect(modules).to.eql({
+				accounts: {}
+			});
+		});
+
+		it('should be okay with correct params', function () {
+			delegate.bind(accountsMock);
+			var modules = Delegate.__get__('modules');
+
+			expect(modules).to.eql({
+				accounts: accountsMock
+			});
+		});
+	});
+
+	describe('calculateFee', function () {
+
+		it('should return the correct fee for delegate transaction', function () {
+			expect(delegate.calculateFee(trs)).to.equal(node.constants.fees.delegate);
+		});
+	});
+
+	describe('verify', function () {
+
+		it('should return error if recipientId exist', function (done) {
```

Comment:
"should return error if recipientId doesn't exists"

Reply:
No, the test description is correct. We return an error when the recipientId exists and not when it doesn't exist